src/activity_handlers.cpp

```diff
@@ -1274,6 +1275,38 @@ void activity_handlers::butcher_finish( player_activity *act, player *p )
     resume_for_multi_activities( *p );
 }
 
+void activity_handlers::milk_finish( player_activity *act, player *p )
+{
+    if( act->coords.empty() ) {
+        debugmsg( "milking activity with no position of monster stored" );
+        return;
+    }
+    const tripoint source_pos = g->m.getlocal( act->coords.at( 0 ) );
+    Creature *c = g->critter_at( source_pos );
+    monster *source_mon = dynamic_cast<monster *>( c );
+    if( source_mon == nullptr ) {
+        debugmsg( "could not find source creature for liquid transfer" );
+        return;
+    }
+    const auto milked_item = source_mon->ammo.find( "milk_raw" );
+    if( milked_item == source_mon->ammo.end() ) {
+        debugmsg( "animal has no milk ammo type" );
+        return;
+    }
+    item milk( milked_item->first, calendar::turn, milked_item->second );
+    milk.set_item_temperature( 311.75 );
+    if( liquid_handler::handle_liquid( milk, nullptr, 1, nullptr, nullptr, -1, source_mon ) ) {
+        source_mon->ammo["milk_raw"] = 0;
```

Comment:
There's already `milked_item->second` for this purpose. It can be made non-const and reused (here and below).